metis-authentication/metis-authentication-service/src/main/java/eu/europeana/metis/authentication/dao/ZohoAccessClientDao.java

```diff
@@ -36,40 +44,79 @@ public ZohoAccessClientDao(String zohoBaseUrl, String zohoAuthenticationToken) {
+            SEARCH_RECORDS_STRING);
     UriComponentsBuilder builder = UriComponentsBuilder.fromHttpUrl(contactsSearchUrl)
         .queryParam(AUTHENTICATION_TOKEN_STRING, zohoAuthenticationToken)
-        .queryParam(SCOPE_STRING, "crmapi")
-        .queryParam(CRITERIA_STRING, String.format("(Email:%s)", email));
+        .queryParam(SCOPE_STRING, CRMAPI_STRING)
+        .queryParam(CRITERIA_STRING, String.format("(%s:%s)", EMAIL_FIELD, email));
 
     RestTemplate restTemplate = new RestTemplate();
     String contactResponse = restTemplate
         .getForObject(builder.build().encode().toUri(), String.class);
     LOGGER.info(contactResponse);
     ObjectMapper mapper = new ObjectMapper();
     JsonNode jsonContactResponse = mapper.readTree(contactResponse);
-    if (jsonContactResponse.get(RESPONSE_STRING).get(RESULT_STRING) == null)
+    if (jsonContactResponse.get(RESPONSE_STRING).get(RESULT_STRING) == null) {
       return null;
-    return jsonContactResponse.get(RESPONSE_STRING).get(RESULT_STRING).get(CONTACTS_MODULE_STRING).get("row").get("FL");
+    }
+    return jsonContactResponse.get(RESPONSE_STRING).get(RESULT_STRING).get(CONTACTS_MODULE_STRING)
+        .get(ROW_STRING).get(FIELDS_LABEL);
   }
 
   public JsonNode getOrganizationByOrganizationName(String organizationName) throws IOException {
     String contactsSearchUrl = String
-        .format("%s/%s/%s/%s", zohoBaseUrl, "json", ACCOUNTS_MODULE_STRING, SEARCH_RECORDS_STRING);
+        .format("%s/%s/%s/%s", zohoBaseUrl, JSON_STRING, ACCOUNTS_MODULE_STRING,
+            SEARCH_RECORDS_STRING);
     UriComponentsBuilder builder = UriComponentsBuilder.fromHttpUrl(contactsSearchUrl)
         .queryParam(AUTHENTICATION_TOKEN_STRING, zohoAuthenticationToken)
-        .queryParam(SCOPE_STRING, "crmapi")
-        .queryParam(CRITERIA_STRING, String.format("(Account Name:%s)", organizationName));
+        .queryParam(SCOPE_STRING, CRMAPI_STRING)
+        .queryParam(CRITERIA_STRING,
+            String.format("(%s:%s)", ORGANIZATION_NAME_FIELD, organizationName));
 
     RestTemplate restTemplate = new RestTemplate();
     String contactResponse = restTemplate
         .getForObject(builder.build().encode().toUri(), String.class);
     LOGGER.info(contactResponse);
     ObjectMapper mapper = new ObjectMapper();
     JsonNode jsonContactResponse = mapper.readTree(contactResponse);
-    if (jsonContactResponse.get(RESPONSE_STRING).get(RESULT_STRING) == null)
+    return findExactMatchOfOrganization(jsonContactResponse, organizationName);
+  }
+
+  private JsonNode findExactMatchOfOrganization(JsonNode jsonOrgizationsResponse,
+      String organizationName) {
+    if (jsonOrgizationsResponse.get(RESPONSE_STRING).get(RESULT_STRING) == null) {
+      return null;
+    }
+    JsonNode organizationJsonNode = jsonOrgizationsResponse.get(RESPONSE_STRING).get(RESULT_STRING)
+        .get(ACCOUNTS_MODULE_STRING).get(ROW_STRING).get(FIELDS_LABEL);
+    if (organizationJsonNode == null) {
```

Comment:
I suggest to use 
jsonOrgizationsResponse.....get(ROW_STRING).isArray() instead of a null comparison to determine whether we are dealing with an object or an array.